AWS AMI misconfiguration prevents QuestDB startup

### To reproduce

1. Set up an EC2 instance with the QuestDB AMI (I tested in us-west-2 using ami-0b327b33c311154aa "questdb-9.3.4-al2023-x86_64-ebs-4f5319bd-6c55-47f6-86d8-c1786f470cc6" , ymmv in other regions with other AMIs)
2. QuestDB doesn't start
3. SSH into instance
4. Find that `java --version` says Java 11; correct it with `sudo yum install java-17-amazon-corretto`
5. Find that `/var/lib/questdb/conf/server.conf` contains a rogue single-quote `'` at the end of file; correct this with `sudo vim`
6. Service now starts properly with `sudo service questdb start`; can be easily verified with `journalctl -e -u questdb.service`

### QuestDB version:

9.3.4

### OS, in case of Docker specify Docker and the Host OS:

Per AMI

### File System, in case of Docker specify Host File System:

ext4

### Full Name:

Jordan Angold

### Affiliation:

Sophos

### Have you followed Linux, MacOs kernel configuration steps to increase Maximum open files and Maximum virtual memory areas limit?

- [x] Yes, I have

### Additional context

I happen to be instantiating this instance through terraform, and have not yet attempted to modify the configuration file beyond the `'` fix mentioned in the repro steps. The instance is a modest `m6i.large` intended for light use; it has been configured with an 8GB root partition and a larger /dev/xvdb data partition, both gp3. 

I did check the `ulimit` values on the host and they are both "unlimited".

Both of these issues seem to have clear sources:
1. https://github.com/questdb/questdb/blob/master/pkg/ami/marketplace/build.bash#L33
2. https://github.com/questdb/questdb/blob/master/pkg/ami/marketplace/assets/server.conf#L1181

Relevant logs from `journalctl`:
```
May 12 18:08:58 hostname-redacted questdb[1929]: Error occurred during initialization of boot layer
May 12 18:08:58 hostname-redacted questdb[1929]: java.lang.module.FindException: Error reading module: /usr/local/bin/questdb.jar
May 12 18:08:58 hostname-redacted questdb[1929]: Caused by: java.lang.module.InvalidModuleDescriptorException: Unsupported major.minor version 61.0
May 12 18:08:58 hostname-redacted systemd[1]: questdb.service: Main process exited, code=exited, status=1/FAILURE
May 12 18:08:58 hostname-redacted systemd[1]: questdb.service: Failed with result 'exit-code'.
May 12 18:09:00 hostname-redacted systemd[1]: /usr/lib/systemd/system/questdb.service:28: Standard output type syslog is obsolete, automatically updating to journal. Please update your unit file, and consider removin>
May 12 18:09:00 hostname-redacted systemd[1]: /usr/lib/systemd/system/questdb.service:29: Standard output type syslog is obsolete, automatically updating to journal. Please update your unit file, and consider removin>
May 12 18:09:00 hostname-redacted systemd[1]: questdb.service: Scheduled restart job, restart counter is at 1.
May 12 18:09:00 hostname-redacted systemd[1]: Stopped questdb.service - QuestDB.
```
and:
```
May 12 18:19:01 hostname-redacted systemd[1]: Starting questdb.service - QuestDB...
May 12 18:19:01 hostname-redacted systemd[1]: Started questdb.service - QuestDB.
May 12 18:19:01 hostname-redacted questdb[29689]: Reading log configuration from /var/lib/questdb/conf/log.conf
May 12 18:19:02 hostname-redacted questdb[29689]: 2026-05-12T18:19:01.684101Z A server-main QuestDB 9.3.4. Copyright (C) 2014-2026, all rights reserved.
May 12 18:19:02 hostname-redacted questdb[29689]: 2026-05-12T18:19:02.122956Z A server-main linux-x86-64 [AVX512,10, 64 bits, 2 processors]
May 12 18:19:02 hostname-redacted questdb[29689]: 2026-05-12T18:19:02.123089Z A server-main fs.file-max checked [limit=134217728]
May 12 18:19:02 hostname-redacted questdb[29689]: 2026-05-12T18:19:02.123299Z A server-main vm.max_map_count checked [limit=134217728]
May 12 18:19:02 hostname-redacted questdb[29689]: 2026-05-12T18:19:02.142912Z I server-main Web Console is up to date
May 12 18:19:02 hostname-redacted questdb[29689]: 2026-05-12T18:19:02.164547Z A server-main Server config: /var/lib/questdb/conf/server.conf
May 12 18:19:02 hostname-redacted questdb[29689]: io.questdb.Bootstrap$BootstrapException: io.questdb.ServerConfigurationException: Configuration issues:
May 12 18:19:02 hostname-redacted questdb[29689]:     Invalid settings (not recognized, probable typos):
May 12 18:19:02 hostname-redacted questdb[29689]:         * '
May 12 18:19:02 hostname-redacted questdb[29689]:     Deprecated settings (recognized but optionally superseded by newer settings):
May 12 18:19:02 hostname-redacted questdb[29689]:         * query.timeout.sec: Replaced by `query.timeout`
May 12 18:19:02 hostname-redacted questdb[29689]:         at io.questdb@9.3.4/io.questdb.Bootstrap.<init>(Bootstrap.java:224)
May 12 18:19:02 hostname-redacted questdb[29689]:         at io.questdb@9.3.4/io.questdb.Bootstrap.<init>(Bootstrap.java:98)
May 12 18:19:02 hostname-redacted questdb[29689]:         at io.questdb@9.3.4/io.questdb.ServerMain.<init>(ServerMain.java:83)
May 12 18:19:02 hostname-redacted questdb[29689]:         at io.questdb@9.3.4/io.questdb.ServerMain.main(ServerMain.java:137)
May 12 18:19:02 hostname-redacted questdb[29689]: Caused by: io.questdb.ServerConfigurationException: Configuration issues:
May 12 18:19:02 hostname-redacted questdb[29689]:     Invalid settings (not recognized, probable typos):
May 12 18:19:02 hostname-redacted questdb[29689]:         * '
May 12 18:19:02 hostname-redacted questdb[29689]:     Deprecated settings (recognized but optionally superseded by newer settings):
May 12 18:19:02 hostname-redacted questdb[29689]:         * query.timeout.sec: Replaced by `query.timeout`
May 12 18:19:02 hostname-redacted questdb[29689]:         at io.questdb@9.3.4/io.questdb.PropServerConfiguration.validateProperties(PropServerConfiguration.java:2389)
May 12 18:19:02 hostname-redacted questdb[29689]:         at io.questdb@9.3.4/io.questdb.PropServerConfiguration.<init>(PropServerConfiguration.java:828)
May 12 18:19:02 hostname-redacted questdb[29689]:         at io.questdb@9.3.4/io.questdb.DynamicPropServerConfiguration.<init>(DynamicPropServerConfiguration.java:151)
May 12 18:19:02 hostname-redacted questdb[29689]:         at io.questdb@9.3.4/io.questdb.DynamicPropServerConfiguration.<init>(DynamicPropServerConfiguration.java:187)
May 12 18:19:02 hostname-redacted questdb[29689]:         at io.questdb@9.3.4/io.questdb.PropBootstrapConfiguration.getServerConfiguration(PropBootstrapConfiguration.java:40)
May 12 18:19:02 hostname-redacted questdb[29689]:         at io.questdb@9.3.4/io.questdb.Bootstrap.<init>(Bootstrap.java:183)
May 12 18:19:02 hostname-redacted questdb[29689]:         ... 3 more
May 12 18:19:02 hostname-redacted systemd[1]: questdb.service: Main process exited, code=exited, status=55/n/a
```

--- a/pkg/ami/marketplace/assets/server.conf
+++ b/pkg/ami/marketplace/assets/server.conf
@@ -1178,4 +1178,3 @@ pg.user=admin
 ## Polling rate of a replica instance to check for new changes (milliseconds).
 #replication.replica.poll.interval=1000
 
-'

--- a/pkg/ami/marketplace/assets/systemd.service
+++ b/pkg/ami/marketplace/assets/systemd.service
@@ -1,16 +1,21 @@
 [Unit]
 Description=QuestDB
 Documentation=https://www.questdb.io/docs/introduction
-After=network.target
+Wants=network-online.target
+After=network-online.target
 RequiresMountsFor=/var/lib/questdb
 
 [Service]
-Type=simple
+Type=exec
 Restart=always
 RestartSec=2
-LimitNOFILE=134217728
 User=questdb
 Group=questdb
+
+# Resource limits
+LimitNOFILE=infinity
+TasksMax=infinity
+
 Environment=QDB_PACKAGE=ami-market
 ExecStart=/usr/bin/java \
     -XX:+UnlockExperimentalVMOptions \
@@ -23,10 +28,11 @@ ExecStart=/usr/bin/java \
     -m io.questdb/io.questdb.ServerMain \
     -d /var/lib/questdb
 ExecReload=/bin/kill -s HUP $MAINPID
+
 # Prevent writes to /usr, /boot, and /etc
 ProtectSystem=full
-StandardOutput=syslog
-StandardError=syslog
+StandardOutput=journal
+StandardError=journal
 SyslogIdentifier=questdb
 
 [Install]

--- a/pkg/ami/marketplace/build.bash
+++ b/pkg/ami/marketplace/build.bash
@@ -30,7 +30,7 @@ QUESTDB_DATA_DIR=/var/lib/questdb
 # Install dependencies
 sudo dnf update -y -q
 sudo dnf install -y -q \
-    java-11-amazon-corretto-headless \
+    java-17-amazon-corretto-headless \
     ec2-instance-connect \
     amazon-ssm-agent \
     amazon-cloudwatch-agent \

--- a/pkg/ami/marketplace/build.bash
+++ b/pkg/ami/marketplace/build.bash
@@ -26,15 +26,27 @@
 
 set -euxo pipefail
 QUESTDB_DATA_DIR=/var/lib/questdb
+GRAALVM_VERSION=25.0.2
+GRAALVM_HOME=/usr/lib/jvm/graalvm-community-java25
+GRAALVM_TARBALL=graalvm-community-jdk-${GRAALVM_VERSION}_linux-x64_bin.tar.gz
 
 # Install dependencies
 sudo dnf update -y -q
 sudo dnf install -y -q \
-    java-17-amazon-corretto-headless \
     ec2-instance-connect \
     amazon-ssm-agent \
     amazon-cloudwatch-agent \
     jq
+
+# Install GraalVM Community JDK 25 and make it the system default JVM.
+wget -q https://github.com/graalvm/graalvm-ce-builds/releases/download/jdk-${GRAALVM_VERSION}/${GRAALVM_TARBALL}
+sudo mkdir -p ${GRAALVM_HOME}
+sudo tar -xzf ${GRAALVM_TARBALL} --strip-components=1 -C ${GRAALVM_HOME}
+rm -f ${GRAALVM_TARBALL}
+sudo alternatives --install /usr/bin/java java ${GRAALVM_HOME}/bin/java 250000
+sudo alternatives --set java ${GRAALVM_HOME}/bin/java
+sudo alternatives --install /usr/bin/javac javac ${GRAALVM_HOME}/bin/javac 250000
+sudo alternatives --set javac ${GRAALVM_HOME}/bin/javac
 
 sudo mv /tmp/scripts/1-per-boot.sh /var/lib/cloud/scripts/per-boot/
 sudo mv /tmp/assets/99-questdb.conf /etc/sysctl.d/

